src/views/credentials/addCredentialModal.tsx

```diff
@@ -26,19 +27,35 @@ interface AddCredentialModalProps {
   onSubmit?: (payload: any) => void;
 }
 
+interface CredentialFormProps extends Omit<AddCredentialModalProps, 'isOpen'> {
+  useForm?: typeof useCredentialForm;
+}
+
 interface CredentialFormType extends Partial<CredentialType> {
   authenticationType?: string;
 }
 
-interface CredentialFormFieldsProps {
-  formData?: CredentialFormType;
-  authType?: string;
-  typeValue?: string;
-  setAuthType?: (credentialType: string) => void;
-  handleInputChange?: (field: string, value: string) => void;
-}
+const getCleanedFormData = (formData, authType) => {
+  const cleanedData = { ...formData };
+
+  if (authType === 'Token') {
+    cleanedData.password = '';
+    cleanedData.username = '';
+  } else if (authType === 'Username and Password') {
+    cleanedData.auth_token = '';
+    cleanedData.ssh_key = '';
+    cleanedData.ssh_passphrase = '';
+  } else if (authType === 'SSH Key') {
+    cleanedData.password = '';
+  }
```

Comment:
**important:** since we're crunched on time let's focus instead on direct changes to align with more with the "addSourceModal" that way we form a more consistent pattern and get this out the door faster... @nicolearagao we can iterate on @mirekdlugosz approach after the release

**minimal required change:**
layered if/else are prone to errors, keep it simple, this needs to be a switch in the form of...
```
switch (authType) {
    case 'Token':
      cleanedData.password = '';
      cleanedData.username = '';
      break;
    case 'Username and Password':
      cleanedData.auth_token = '';
      cleanedData.ssh_key = '';
      cleanedData.ssh_passphrase = '';
      break;
    case 'SSH Key':
      cleanedData.password = '';
      break;
  }
```